config-collector/src/main/java/no/sysco/middleware/kafka/interceptor/config/BaseConfigCollectorInterceptor.java

```diff
@@ -0,0 +1,111 @@
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static no.sysco.middleware.kafka.interceptor.config.ConfigCollectorInterceptorConfig.CONFIG_COLLECTOR_BLACKLIST_DEFAULT;
+
+/**
+ * @author Sysco Middleware AS
+ */
+class BaseConfigCollectorInterceptor implements Configurable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BaseConfigCollectorInterceptor.class.getName());
+
+    private final ClientConfig.ClientType clientType;
+
+    BaseConfigCollectorInterceptor(ClientConfig.ClientType clientType) {
+        this.clientType = clientType;
+    }
+
+    /**
+     * Callback interceptor function
+     *
+     * @param configs user provided configuration properties
+     */
+    @Override
+    public void configure(final Map<String, ?> configs) {
+        try {
+            final ProducerRecord<String, byte[]> record = buildConfigRecord(configs);
+            final Producer<String, byte[]> producer = buildConfigProducer(configs);
+            producer.send(record).get();
+            producer.close();
+        } catch (Exception ex) {
+            LOGGER.error("Config Collector Interceptor failed", ex);
+        }
+    }
+
+    private Producer<String, byte[]> buildConfigProducer(Map<String, ?> configs) {
+        final HashMap<String, Object> configProducerConfigs = new HashMap<>();
+        configs.forEach((k, v) -> {
+            if (!k.contains(ProducerConfig.INTERCEPTOR_CLASSES_CONFIG)) {
+                configProducerConfigs.put(k, v);
+            }
+        });
+        configProducerConfigs.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        configProducerConfigs.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class.getName());
+
+        return new KafkaProducer<>(configProducerConfigs);
+    }
+
+    private ProducerRecord<String, byte[]> buildConfigRecord(Map<String, ?> configs) {
+        final String topicName = configTopic(configs);
+        final String clientId = (String) configs.get("client.id");
+        final ClientConfig.Builder builder = ClientConfig.newBuilder();
+        builder.setType(clientType).setId(clientId);
+        configs.forEach((k, v) -> {
+            if (CONFIG_COLLECTOR_BLACKLIST_DEFAULT.contains(k)) {
+                builder.putEntries(k, "<value>");
+            } else {
+                builder.putEntries(k, v.toString());
+            }
+        });
+        final String key = String.format("%s-%s", clientType.name(), clientId);
+        final byte[] value = builder.build().toByteArray();
+        return new ProducerRecord<>(topicName, key, value);
+    }
+
+    private String configTopic(Map<String, ?> configs) {
+        final String topicName;
+        if (configs.containsKey(ConfigCollectorInterceptorConfig.CONFIG_COLLECTOR_TOPIC_CONFIG)) {

```

Comment:
Could we just use config.get() and check for null value instead of use containsKey & get (two map lookups) ?